core/types/src/account.rs

```diff
@@ -0,0 +1,163 @@
+/// keys.
+#[derive(Debug, Zeroize)]
+pub struct Account {
+    /// Root view private key
+    view_private: RootViewPrivate,
+    /// Root spend private key
+    spend_private: RootSpendPrivate,
+}
+
+impl Account {
+    /// Create an account from existing private keys
+    pub fn new(view_private: RootViewPrivate, spend_private: RootSpendPrivate) -> Self {
+        Self {
+            view_private,
+            spend_private,
+        }
+    }
+
+    /// Fetch account view public key
+    pub fn view_public_key(&self) -> RootViewPublic {
+        RootViewPublic::from(&self.view_private)
+    }
+
+    /// Fetch account spend public key
+    pub fn spend_public_key(&self) -> RootSpendPublic {
+        RootSpendPublic::from(&self.spend_private)
+    }
+
+    /// Fetch account view private key
+    pub fn view_private_key(&self) -> &RootViewPrivate {
+        &self.view_private
+    }
+
+    /// Fetch account spend private key
+    pub fn spend_private_key(&self) -> &RootSpendPrivate {
+        &self.spend_private
+    }
+}
+
+/// Mobilecoin private subaddress object
+#[derive(Clone, Debug, PartialEq)]
+pub struct SpendSubaddress {
+    /// sub-address view private key
+    pub view_private: SubaddrViewPrivate,
+    /// sub-address spend private key
+    pub spend_private: SubaddrSpendPrivate,
+}
+
+impl RingCtAddress for SpendSubaddress {
+    /// Fetch view public address
+    fn view_public_key(&self) -> SubaddrViewPublic {
+        SubaddrViewPublic::from(&self.view_private)
+    }
+
+    /// Fetch spend public address
+    fn spend_public_key(&self) -> SubaddrSpendPublic {
+        SubaddrSpendPublic::from(&self.spend_private)
+    }
+}
+
+impl SpendSubaddress {
+    /// Fetch subaddress view private key
+    pub fn view_private_key(&self) -> &SubaddrViewPrivate {
+        &self.view_private
+    }
+
+    /// Fetch subaddress spend private key
+    pub fn spend_private_key(&self) -> &SubaddrSpendPrivate {
+        &self.spend_private
+    }
+}
+
+/// Mobilecoin view-only subaddress object
+#[derive(Clone, Debug, PartialEq)]
+pub struct ViewSubaddress {
+    /// sub-address view private key
+    pub view_private: SubaddrViewPrivate,
+    /// sub-address spend private key
+    pub spend_public: SubaddrSpendPublic,
+}
+
+impl RingCtAddress for ViewSubaddress {
+    /// Fetch view public address
+    fn view_public_key(&self) -> SubaddrViewPublic {
+        SubaddrViewPublic::from(&self.view_private)
+    }
+
+    /// Fetch spend public address
+    fn spend_public_key(&self) -> SubaddrSpendPublic {
+        self.spend_public.clone()
+    }
+}
+
+impl ViewSubaddress {
+    /// Fetch subaddress view private key
+    pub fn view_private_key(&self) -> &SubaddrViewPrivate {
+        &self.view_private
+    }
+}
+
+/// Mobilecoin public subaddress object
+#[derive(Clone, Debug, PartialEq)]
+pub struct PublicSubaddress {
+    /// Public address view public key
+    pub view_public: SubaddrViewPublic,
+    /// Public address spend public key
```

Comment:
:upside_down_face: the "public <blah> public" reads odd
```suggestion
    /// Subaddress view public key
    pub view_public: SubaddrViewPublic,
    /// Subaddress spend public key
```